### PR TITLE
Ncnews 6

### PR DESCRIPTION
**Pull request by H-Shihaibi**
code review nc news tickets 5-6 - mistakenly didn't add 5 previously

### CODE REVIEW

**Review by HMelesi — CHANGES_REQUESTED** [review]
Task 5: GET /api/users - all approved, well done Hamam :) 

Task 6: PATCH /api/articles/:article_id - quite a few changes requested on this one, have a look at my comments and let me know if anything is unclear.

File: __tests__/app.test.js [hunk 1]
```diff
@@ -28,3 +28,104 @@ describe("GET /api/topics", () => {
...
+          created_at: "2020-10-16T05:03:00.000Z",
+          votes: 0,
+        });
+        expect().toBe();
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply __tests__/app.test.js:48]: Deleted, thanks!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -28,3 +28,104 @@ describe("GET /api/topics", () => {
...
+  });
+  test("status:400, responds with a invalid input", () => {
+    const ARTICLE_ID = "not a number";
+    return request(app)
+      .get(`/api/articles/${ARTICLE_ID}`)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toEqual("Invalid data type");
+      });
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply __tests__/app.test.js:59]: Done!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -28,3 +28,104 @@ describe("GET /api/topics", () => {
...
+          );
+        });
+      });
+  });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  test("201: update votes count, respond with newly updated article", () => {
+    const updateVotes = {
+      votes: +1,
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply __tests__/app.test.js:88]: Removed.

[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply __tests__/app.test.js:88]: Removed

File: __tests__/app.test.js [hunk 4]
```diff
@@ -28,3 +28,104 @@ describe("GET /api/topics", () => {
...
+    };
+    return request(app)
+      .patch("/api/articles/3")
+      .send(updateVotes)
+      .expect(200)
+      .then((res) => {
+        const { body } = res;
+        expect(body.article).toEqual({
+          ...body.article,
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply __tests__/app.test.js:97]: Good shout! I've added an object const to the test and replaced the (...body.article).

File: __tests__/app.test.js [hunk 5]
```diff
@@ -28,3 +28,104 @@ describe("GET /api/topics", () => {
...
+      votes: -99,
+    };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(updateVotes)
+      .expect(200)
+      .then((res) => {
+        const { body } = res;
+        expect(body.article).toEqual({
+          ...body.article,
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply __tests__/app.test.js:113]: Done

File: __tests__/app.test.js [hunk 6]
```diff
@@ -28,3 +28,104 @@ describe("GET /api/topics", () => {
...
+        });
+        expect(body.article.votes).toBe(1);
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply __tests__/app.test.js:115]: Noted

File: controllers/articles-controller.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  const articleId = req.params.article_id;
+  selectArticleById(articleId)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch(next);
+};
+
+exports.patchArticleVotesById = (req, res, next) => {
+  const input = req.params.article_id;
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply controllers/articles-controller.js:16]: Done

File: controllers/articles-controller.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  const newVotesCount = req.body;
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply controllers/articles-controller.js:17]: Good note - All done, to fail and work.

File: models/articles.model.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+exports.selectArticleById = (articleId) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1;", [articleId])
+    .then((article) => {
+      return article.rows[0];
+    });
+};
+
+exports.updateArticleVotesById = (articleId, newVotesCount) => {
+  const { votes } = newVotesCount;
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply models/articles.model.js:12]: done

File: models/articles.model.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  const arrValues = [votes, articleId];
```
[the remark this answers is not in the document]
Comment (H-Shihaibi) [reply models/articles.model.js:13]: Thanks!

**Review by HMelesi — APPROVED** [review]

**Pull request merged**